### PR TITLE
stabilize abi_thiscall

### LINKED ISSUE #42202: Tracking issue for the `thiscall` calling convention
Added in #42058, which fixes #42044 and related requests from other bugs.

This calling convention is the default calling convention for C++ instance methods in the (MSVC) Windows ABI.  At the very least, bindgen (and its dependency syntex) need this to exist so they have a reasonable chance of representing C++ instance methods in an AST.  `thiscall` is therefore important for the interop story between Rust and C++ on Windows.

I don't *think* you could actually write callable C++ instance methods in Rust for MSVC, since mangled names in MSVC contain characters that, AFAICT, are not legal in Rust identifiers ('?', for instance).  But maybe there is cleverness around assigning assembly names to things that I am not aware of.

### PR DESCRIPTION
**Pull request by Trolldemorted**
Closes https://github.com/rust-lang/rust/issues/42202, stabilizing the use of the "thiscall" ABI.

FCP was substituted by a poll, and the poll has been accepted.

### CODE REVIEW

File: src/doc/unstable-book/src/language-features/abi-thiscall.md [hunk 1]
```diff
@@ -1,12 +0,0 @@
...
-The tracking issue for this feature is: [#42202]
-
-[#42202]: https://github.com/rust-lang/rust/issues/42202
-
-------------------------
-
-The MSVC ABI on x86 Windows uses the `thiscall` calling convention for C++
-instance methods by default; it is identical to the usual (C) calling
-convention on x86 Windows except that the first parameter of the method,
-the `this` pointer, is passed in the ECX register.
```
Comment (oli-obk) [inline src/doc/unstable-book/src/language-features/abi-thiscall.md:12]: Is this still documented somewhere? Do we have a section for ABIs in the reference?

**Comment by oli-obk** [comment]
@bors r+

Thanks!

**Comment by kadiwa4** [comment]
Shouldn't the version string be updated when moving a feature to `accepted`?

**Comment by Trolldemorted** [comment]
I read https://rustc-dev-guide.rust-lang.org/stabilization_guide.html#updating-the-feature-gate-listing and it didn't explicitly tell me to change the version, did I misunderstand that? Since there was a version string present, I didn't change because the example has "CURRENT_RUSTC_VERSION" in both active and accepted.

**Comment by kadiwa4** [comment]
That example confuses me too (why would the version string say `CURRENT_RUSTC_VERSION` *before* being accepted?)

Still, I think you're supposed to update that, other stabilization PRs do that too https://github.com/rust-lang/rust/pull/109677/files#diff-307e0d3a2037c11a3fa16822fbaa0fec08e57ac7d0d6e7354f6005c9482a9e26

**Pull request merged**